app/views.py

```diff
@@ -0,0 +1,131 @@
+
+    # create an admin user and add to database
+    encrypted_password = hash_password('password')
+    if not user_datastore.get_user('admin@mporter.co'):
+        user_datastore.create_user(email='admin@mporter.co', password=encrypted_password)
+
+    db_init.session.commit()
+
+    # make admin@mporter.co the admin user
+    user_datastore.add_role_to_user('admin@mporter.co', 'admin')
+    db_init.session.commit()
+
+
+@app.route('/')
+def index():
+    """
+    serves the static index page with features and stuff
+    """
+    return render_template('index.html')
+
+
+@app.route('/mentee')
+@login_required
+def mentee():
+    """
+    serves a page with tasks by the mentee and a form to add new tasks
+    additionally a button to logout. KISS.
+    """
+    current_user_id = current_user.get_id()
+    user = user_datastore.get_user(current_user_id)
+
+    is_admin = False
+    if len(user.roles) > 0:
+        is_admin = True if user.roles[0].name == 'admin' else False
+
+    from .models import Tasks, Mentees
+
+    user_tasks = Tasks\
+        .query\
+        .filter_by(mentee_id=user.id) \
+        .filter(Tasks.at_created >= datetime.today() - timedelta(days=1))\
+        .all()
+
+    mentee_obj = Mentees.query.filter_by(id=user.id).first()
+
+    user_obj = {
+        'user_id': user.id,
+        'user_email': user.email,
+        'user_tasks': user_tasks,
+        'user_mentors': mentee_obj.mentor,
+        'is_admin': is_admin
+    }
+
+    return render_template('mentee.html', user=user_obj)
+
+
+@app.route('/new-task', methods=['POST'])
+@login_required
+def new_task():
+    """
+    An api to add a new task. API needs authentication, and the task will be added to the authenticated user's tasks.
+    @:param task: text of the task to be added in post request body
+    """
+    task = request.form.get('task')
+    current_user_id = current_user.get_id()
+    user = user_datastore.get_user(current_user_id)
+
+    from .models import Tasks
+
+    task = Tasks(mentee_id=user.id, task=task)
+
+    db_init.session.add(task)
+    db_init.session.commit()
+
+    return redirect(url_for('mentee'))
+
+
+@app.route('/new-mentor', methods=['POST'])
+@login_required
+def new_mentor():
+    """
+    add a mentor to user's mentors list. If mentor is not present, add a new mentor as well with the name supplied
+    note that if a mentor is present, the mentor_name supplied by user is ignored
+    @:param: mentor-name, mentor-email
+    """
+    from .models import Mentors, Mentees
+
+    mentor_name = request.form.get('mentor-name')
+    mentor_email = request.form.get('mentor-email')
+
+    # something's fishy, return
+    if not mentor_name or not mentor_email:
+        return redirect(url_for('mentee'))
+
+    current_user_id = current_user.get_id()
+    mentee_present = Mentees.query.filter_by(id=current_user_id).first()
+
+    if not mentee_present:
+        raise KeyError('mentee {} is not present in the database'.format(current_user_id))
```

Comment:
No generic errors from controller layer
http://flask.pocoo.org/docs/1.0/patterns/apierrors/

Reply:
Ohh, something like custom exception classes. Thik hai.

Reply:
https://stackoverflow.com/questions/29332056/global-error-handler-for-any-exception

https://opensource.com/article/17/3/python-flask-exceptions

https://rock-it.pl/custom-exception-handler-in-django/

https://medium.com/@mwhitt.w/restful-error-messages-with-django-537047892dff


Another way would be all 

if using Class based views, it should inherit from a base class which in turn is inheriting different classes which handle things such as logging, exception handling, common errors (record does not exist) and handles them efficiently in the parent class.

in function based views, create and stack decorator for the same (wrap the function)
https://stackoverflow.com/questions/739654/how-to-make-a-chain-of-function-decorators#1594484

Reply:
Aaishappat. Thik hai. That's a lot of data to ingest :D Will read up. Thanks.